components/webvr_traits/lib.rs

```diff
@@ -30,3 +30,9 @@ pub use rust_webvr_api::VRLayer as WebVRLayer;
 pub use rust_webvr_api::VRMainThreadHeartbeat as WebVRMainThreadHeartbeat;
 pub use rust_webvr_api::VRPose as WebVRPose;
 pub use rust_webvr_api::VRStageParameters as WebVRStageParameters;
+
+#[derive(Deserialize, Serialize)]
+pub struct WebVRPoseInformation {
+    pub frame: WebVRFutureFrameData,
+    pub gamepads: Vec<(u32, WebVRGamepadState)>,
```

Comment:
nit: A comment about what that `u32` is for would be nice.

Reply:
The frame is a asynchronous, but the gamepad state is synchronous, which seems odd, at least worth commenting on. IIRC the frame is async to deal with the case that the frame data is only available on the main thread, to avoid having the webgl thread block on the main thread, since that can cause deadlock. Are there platforms where the gamepad state is also only available on the main thread?

Reply:
Not that I know of.

Reply:
I think the magicleap assumes that gamepad state is queried on the main thread, but I'm not sure if this is a requirement, or just the way their example code is written.